src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
+      const newComment = {
+        name: name.trim(),
+        email: email.trim(),
+        body: body.trim(),
+        postId: selectedPost?.id,
```

Comment:
Issue: The optional chaining operator `?.` is not necessary here since `selectedPost` is already checked for truthiness. You can safely use `selectedPost.id`.